elastic/integrations
# [checkpoint_email] Initial release of the Check Point Harmony Email & Collaboration

## Proposed commit message

Create New integration package checkpoint_email.

- Added event data stream.
- Added data collection logic for the data stream.
- Added the ingest pipeline for the data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboard and visualizations.
- Added pipeline test for the data stream.
- Added system test cases for the data stream.



## Checklist


## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/checkpoint_email directory.
- Run the following command to run tests.
> elastic-package test
```
--- Test results for package: checkpoint_email - START ---
╭──────────────────┬─────────────┬───────────┬───────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME                                 │ RESULT │ TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼───────────────────────────────────────────┼────────┼──────────────┤
│ checkpoint_email │ event       │ pipeline  │ (ingest pipeline warnings test-event.log) │ PASS   │   309.6712ms │
│ checkpoint_email │ event       │ pipeline  │ test-event.log                            │ PASS   │ 247.700215ms │
╰──────────────────┴─────────────┴───────────┴───────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: checkpoint_email - END   ---
Done
--- Test results for package: checkpoint_email - START ---
╭──────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ checkpoint_email │ event       │ system    │ default   │ PASS   │ 41.094371033s │
╰──────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: checkpoint_email - END   ---
Done
--- Test results for package: checkpoint_email - START ---
╭──────────────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ checkpoint_email │ event       │ static    │ Verify sample_event.json │ PASS   │ 139.139436ms │
╰──────────────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: checkpoint_email - END   ---
Done
--- Test results for package: checkpoint_email - START ---
╭──────────────────┬─────────────┬───────────┬───────────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME                                                                 │ RESULT │ TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼───────────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ checkpoint_email │             │ asset     │ dashboard checkpoint_email-3f1885e2-633d-4aa9-bf77-56a98d484692 is loaded │ PASS   │      1.189µs │
│ checkpoint_email │             │ asset     │ search checkpoint_email-83662506-f7ee-40ba-9f16-65da29ec7ef4 is loaded    │ PASS   │        300ns │
│ checkpoint_email │ event       │ asset     │ index_template logs-checkpoint_email.event is loaded                      │ PASS   │        273ns │
│ checkpoint_email │ event       │ asset     │ ingest_pipeline logs-checkpoint_email.event-0.1.0 is loaded               │ PASS   │        180ns │
╰──────────────────┴─────────────┴───────────┴───────────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: checkpoint_email - END   ---
Done
```




## Related issues


- Closes #11178 

## Screenshots

![Integration Page](https://github.com/user-attachments/assets/dc1184cd-40c5-4c04-8971-259544883285)
![Overview Page](https://github.com/user-attachments/assets/1896283e-e4e7-43de-adaa-76fa2cf1c173)
